File: collector/src/stock_collector/core/domain/stock.py
"""股票领域模型。"""

from dataclasses import dataclass
from datetime import datetime
from decimal import Decimal


@dataclass(frozen=True)
class Stock:
    """股票领域实体。"""

    symbol: str
    name: str
    market: str
    current_price: Decimal | None = None
    change_percent: Decimal | None = None
    updated_at: datetime | None = None

    def __post_init__(self) -> None:
        if not self.symbol:
            raise ValueError("股票代码不能为空")
        if not self.name:
            raise ValueError("股票名称不能为空")

File: backend/src/main/java/org/cwowhappy/securityanalyze/stock/domain/model/Stock.java
package org.cwowhappy.securityanalyze.stock.domain.model;

import lombok.Builder;
import lombok.Getter;

import java.math.BigDecimal;
import java.time.LocalDateTime;

/**
 * 股票领域实体（Aggregate Root）。
 * 纯 POJO，无框架依赖。
 */
@Getter
@Builder
public class Stock {

    private final StockId id;
    private String symbol;
    private String name;
    private String market;
    private BigDecimal currentPrice;
    private BigDecimal changePercent;
    private LocalDateTime updatedAt;

    /**
     * 更新价格。
     */
    public void updatePrice(BigDecimal newPrice, BigDecimal newChangePercent) {
        if (newPrice == null || newPrice.compareTo(BigDecimal.ZERO) < 0) {
            throw new IllegalArgumentException("价格不能为负数");
        }
        this.currentPrice = newPrice;
        this.changePercent = newChangePercent;
        this.updatedAt = LocalDateTime.now();
    }
}
